### DISCUSSION TITLE
Search and add directly from Open Library ID

### DISCUSSION THREAD

**Discussion by manoelpqueiroz** [mateusz-bak/openreads#773] [Ideas]
**Is your feature request related to a problem? Please describe.**

One of the things that makes organisation more difficult with Openreads is that if you search a book in Open Library by ISBN and get a match, it won't necessarily add the ISBN edition searched in the box, prompting the user to either add a random edition or choose one from Open Library.

However, for very popular books, searching for the desired edition is a nearly impossible task, because the app attempts loading those editions one by one, making loading times increasingly longer and not performatic. This has always ended with me giving up on adding the book entirely.

**Describe the solution you'd like**

The solution I propose is to enable search and book inclusion directly from the Open Library ID, since it is different for every edition available. By enabling the user to directly inform Openreads which version they want to retrieve, it becomes a much easier task.

**Comment by pandalamp** [comment]
Supporting this! If the ISBN can't be used in a search query to match against specific editions, the Open Library ID is the next best thing.

Anyone who thinks this isn't a problem and has a free afternoon is welcome to try adding [this specific edition](https://openlibrary.org/books/OL26825157M/) to Openreads without doing it manually. XD

**Comment by nicolas-robette** [comment]
I'm totally supporting this enhancement as well !

**Comment by mateusz-bak** [comment]
Getting the edition by ISBN is fixed now.

I will implement looking up edition by Open Library ID now.
